aiohttp/connector.py

```diff
@@ -480,7 +480,10 @@ async def connect(self, req: 'ClientRequest',
 
                 raise e
             finally:
-                if not waiters:
+                # `waiters` may be deleted from `self._waiters` by another
+                # coroutine, and `self._waiters[key]` may contain another
+                # deque, the later should not be deleted.
```

Comment:
I guess that the issue is that we are caching the `self._waiters[key]` at the local level by using the local variable `waiters` what about simply doing a check with using the none cached version, like:

```python
if key in self._waiters and not self._waiters[key]:
    ...
```
Maybe it's more clear.

WDYT?

Reply:
It's a good idea. Thanks!